### PR TITLE
Added user's statistics, fixed absentees and attachments

### PR DESCRIPTION
**Pull request by jpalaz**
Sorry for such mess. I'll try to push better next time.

### CODE REVIEW

File: src/main/java/com/exadel/access/control/TrainerControl.java [hunk 1]
```diff
@@ -27,46 +28,73 @@ public TrainerControl(UserService userService, TrainingService trainingService,
         this.entryService = entryService;
     }
 
-    public boolean isCoach(long id) {
-        if (userService.getCurrentId()==trainingService.getTrainerId(id))
+    public boolean isTrainerByTrainingId(long id) {
```
Comment (kmarmaliykov) [inline src/main/java/com/exadel/access/control/TrainerControl.java]: why do we need ByTrainingId here? It is clear from parameter

Same problem below in this class

File: src/main/java/com/exadel/controller/FileUpDownLoadController.java [hunk 2]
```diff
@@ -93,9 +96,9 @@ public FileLoadStatus handleFileDownload(@RequestParam String id, HttpServletReq
                 if (null != outStream)
                     outStream.close();
             } catch (IOException e) {
-                return FileLoadStatus.DOWNLOAD_CLOSING_FAIL;
+                //return FileLoadStatus.DOWNLOAD_CLOSING_FAIL;
             }
-            return FileLoadStatus.DOWNLOAD_SUCCESS;
+            //return FileLoadStatus.DOWNLOAD_SUCCESS;
```
Comment (kmarmaliykov) [inline src/main/java/com/exadel/controller/FileUpDownLoadController.java]: remove this line if you are sure that method should be void

File: src/main/java/com/exadel/controller/FileUpDownLoadController.java [hunk 3]
```diff
@@ -93,9 +96,9 @@ public FileLoadStatus handleFileDownload(@RequestParam String id, HttpServletReq
                 if (null != outStream)
                     outStream.close();
             } catch (IOException e) {
-                return FileLoadStatus.DOWNLOAD_CLOSING_FAIL;
+                //return FileLoadStatus.DOWNLOAD_CLOSING_FAIL;
```
Comment (kmarmaliykov) [inline src/main/java/com/exadel/controller/FileUpDownLoadController.java]: do not ignore exceptions!

File: src/main/java/com/exadel/access/control/TrainerControl.java (hunk 1 above)
Comment (jpalaz) [inline src/main/java/com/exadel/access/control/TrainerControl.java]: No, I supposed, that parameter - trainer ID, but it is training ID. So I renamed all methods for clearness.

File: src/main/java/com/exadel/export/ExportTrainingToPdf.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class ExportTrainingToPdf {
+    private static String FILE =  System.getProperty("user.dir") + File.separator;
```
Comment (kmarmaliykov) [inline src/main/java/com/exadel/export/ExportTrainingToPdf.java]: bad class name. This name is fine for function, not for class

File: src/main/java/com/exadel/access/control/TrainerControl.java (hunk 1 above)
Comment (kmarmaliykov) [inline src/main/java/com/exadel/access/control/TrainerControl.java]: ok, then you should rename parameter, but not method!

File: src/main/java/com/exadel/service/impl/TrainingServiceImpl.java [hunk 5]
```diff
@@ -51,7 +52,7 @@ public Training getTrainingById(long id) {
 
     }
 
-    public long getTrainerId(long id) {
+    public long getTrainerByTrainingId(long id) {
```
Comment (kmarmaliykov) [inline src/main/java/com/exadel/service/impl/TrainingServiceImpl.java]: according to method name Trainer should be returned

**Pull request closed without merging**